src/base/static/js/views/geometry-editor-view.js

```diff
@@ -0,0 +1,583 @@
+    this.map.on("draw:drawvertex", function(evt) {
+      self.numVertices++;
+
+      if (self.layerType === "polygon" && self.numVertices <= 2) {
+        self.displayHelpMessage("draw-polygon-continue-msg");
+      } else if (self.layerType === "polygon" && self.numVertices > 2) {
+        self.displayHelpMessage("draw-polygon-finish-msg");
+      } else if (self.layerType === "polyline" && self.numVertices === 1) {
+        self.displayHelpMessage("draw-polyline-continue-msg");
+      } else if (self.layerType === "polyline" && self.numVertices > 1) {
+        self.displayHelpMessage("draw-polyline-finish-msg");
+      }
+    });
+
+    this.map.on("draw:created", function(evt) {
+      self.resetWorkingGeometry();
+      self.setColorpicker();
+      self.hideIconToolbar();
+
+      self.geometryType = evt.layerType;
+      generateGeometry(evt.layer);
+      self.editingLayerGroup.addLayer(evt.layer);
+      self.swapToolbarVisibility();
+      self.displayHelpMessage("modify-geometry-msg");
+    });
+
+    this.map.on("draw:editvertex", function(evt) {
+      if (self.placeDetailView) {
+        self.placeDetailView.onModified();
+      }
+    });
+
+    this.map.on("draw:editmove", function(evt) {
+      if (self.placeDetailView) {
+        self.placeDetailView.onModified();
+      }
+    });
+
+    this.map.on("draw:edited", function(evt) {
+      evt.layers.eachLayer(function(layer) {
+        
+        // Really there's only one layer to iterate over here, because we have at
+        // most one layer in the editing layer group.
+        generateGeometry(layer);
+      });
+    });
+
+    return this;
+  },
+
+  // ========== Toolbar handlers ==========
+  onClickColorEditModeChange: function(evt) {
+    evt.preventDefault();
+
+    var editMode = $(evt.target).data("edit-mode");
+
+    $(evt.target)
+      .addClass("sp-selected")
+      .siblings()
+      .removeClass("sp-selected");
+    
+    this.colorpickerSettings.editMode = editMode;
+    this.updateColorpicker();
+  },
+
+  onClickColorpicker: function(evt) {
+    evt.preventDefault();
+
+    this.saveWorkingGeometry();
+    this.updateColorpicker();
+    $(".sp-container")
+      .css("left", (evt.pageX - 100) + "px")
+      .css("top", (evt.pageY + 30) + "px")
+    $(".sp-picker-container").toggleClass("hidden");
+    this.setGeometryToolbarHighlighting(evt.currentTarget);
+    this.setDefaultCursor();
+    this.isEditing = false;
+  },
+
+  onClickCreateMarker: function(evt) {
+    evt.preventDefault();
+
+    // Prevent repeat clicks on the same geometry drawing tool
+    if (this.layerType === "marker") return;
+
+    this.resetWorkingGeometry();
+    this.showIconToolbar();
+    this.iconUrl = this.$el.find(".geometry-toolbar-icon-field input:checked").val();
+
+    this.workingGeometry = new L.Draw.Marker(this.map, {
+      icon: new L.icon({
+        iconUrl: this.iconUrl,
+        
+        // TODO: these hard-coded values won't work for all icon types...
```

Comment:
No, but I think I'm going to leave this as-is and make the icon parameters configurable in a later iteration of the editor. For now, we'll just have a requirement that icons in the editor be round.